src/platform/common/utils/multiStepInput.ts

```diff
@@ -224,9 +224,15 @@ export class MultiStepInput<S> implements IMultiStepInput<S> {
                     }),
                     input.onDidHide(() => {
                         (async () => {
-                            reject(
-                                shouldResume && (await shouldResume()) ? InputFlowAction.resume : InputFlowAction.cancel
-                            );
+                            // If we are busy we might be validating, which might pop up new UI like the password UI, which triggers a hide here
+                            // In that case don't reject, promise can wait and continue after validation is done
+                            if (!input.busy) {
```

Comment:
So the issue was throwing me a bit at first. Stuck a catch around our input.showInputBox call in serverSelector.ts since it wasn't returning and saw that something was getting thrown, but not an Error, but an InputFlowAction.
<img width="705" alt="image" src="https://user-images.githubusercontent.com/812783/170554937-f16d7d9a-1943-4caa-b8db-844b8f4642d7.png">
Recently we changed our validation to happen when the URL was committed. For the password case, this validation can pop up another input and hide gets called on the server selection in progress. Then the InputFlowAction gets chucked as an error since the multistep input box uses that for navigation. Fix here was to check for the busy case (which only happens while validating) then in that case don't reject the promise if we get hidden.